lib/MonadEq_Lemmas.thy

```diff
@@ -0,0 +1,278 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+(* Lemmas that support the monad_eq method. Either things that go into the [monad_eq] set
+   or that are needed for such lemmas. Usually about "snd (f s)" or "x \<in> fst (f s)" where f
+   is a nondet monad.
+
+   If you are planning to use the monad_eq method, this is the theory you should import. *)
+
+theory MonadEq_Lemmas
+  imports MonadEq
+begin
+
+lemma snd_return [monad_eq]:
```

Comment:
is it worth it to tweak the style in this theory to remove the space before `[monad_eq]`? should be a simple regexp

Reply:
I've tried not to do that to keep the diff to actual moves. I failed with that plan for Injection_Handler, but for the rest it should have worked. We could do a new PR with a bunch of cleanup, though.

Reply:
Ok, I've changed my mind and applied it directly. All those spaces are gone now.